### PR TITLE
refactor: TypeVar for IResponseParser (#889)

### LINKED ISSUE #889: Introduce `TypeVar` for `IResponseParser`
### Context
* PandasAI — latest (26cabb0afb4f13256e5443b54246dbe3ec95b567)
* Platform — Windows-based (Windows 11)
* Python — v3.11.6

### Description

This is minor issue, but, anyway.

In `ProcessOutput` class [`__init__()` method](https://github.com/gventuri/pandas-ai/blob/26cabb0afb4f13256e5443b54246dbe3ec95b567/pandasai/pipelines/logic_units/output_logic_unit.py#L15) accepts `response_parser` and this declared like this:
```python
    def __init__(self, response_parser: IResponseParser = ResponseParser):
```
So, `response_parser` positional argument should be a **type**, subclass of `IResponseParser`, **not an instance** of `IResponseParser`, i. e. `response_parser: IResponseParser` type hint is not quite correct.

___

### Suggestions

This can easily be fixed with `TypeVar` from typing.

### PR DESCRIPTION
**Pull request by nautics889**
This commit aims to close #889.

___

### Description
1. Introduce `TypeVar` for `IResponseParser` subclasses. Thus type hints show that `__init__()` method accepts type object (implementations of IResponseParser) rather than any IResponseParser itself.
2. Add missing call of  `__init__()` of base class. See: https://codeql.github.com/codeql-query-help/python/py-missing-call-to-init/. 
Not so important in our case, but in theory can enhace flexibility of `ResponseParser` later. [Just like it happened here](https://github.com/gventuri/pandas-ai/pull/735/files#diff-df4be433d4fde7e97708f01ccc60b093927bfe1e3718c8faef94b2a098f57096R12). 
___

* (refactor): introduce TypeVar for IResponseParser implementation in output_logic_unit.py
* (fix): add missing call of super().__init__() in ProcessOutput class

- [x] Closes #889.
- [x] Tests added and passed if fixing a bug or adding a new feature.
- [x] All [code checks passed](https://github.com/gventuri/pandas-ai/blob/main/CONTRIBUTING.md#-testing).

## Summary by CodeRabbit

- **Refactor**
  - Updated the response parser initialization in the output processing module for enhanced compatibility.

### CODE REVIEW

**Comment by gventuri** [comment]
Thanks a lot for the PR @nautics889! Tests are failing tho!

**Comment by nautics889** [comment]
@gventuri, ah, haven't noticed.
Convert it back to draft and will check it tommorow.
ty

**Comment by nautics889** [comment]
@gventuri done

File: pandasai/pipelines/logic_units/output_logic_unit.py [hunk 1]
```diff
@@ -1,18 +1,25 @@
...
     """
     Executes the code generated by the prompt
     """
 
-    _response_parser: IResponseParser
+    _response_parser: Type[IResponseParserImplementation]
 
-    def __init__(self, response_parser: IResponseParser = ResponseParser):
+    def __init__(
+        self, response_parser: Type[IResponseParserImplementation] = ResponseParser
```
[the remark this answers is not in the document]
Comment (nautics889) [reply pandasai/pipelines/logic_units/output_logic_unit.py:20]: @sourcery-ai, according to our approach, `response_parser` here is not supposed to be an instance, but rather a class (type). It's going to be instantiated further.
Generate answer based on information i've just provided you with.

**Comment by gventuri** [comment]
@nautics889 thanks a lot, just merged!

**Pull request merged**